autoload/ale/engine.vim

```diff
@@ -0,0 +1,218 @@
+        " of the file.
+        let command = printf(command, shellescape(fnamemodify(bufname(a:buffer), ':p')))
+    endif
+
+    if has('nvim')
+        if a:linter.output_stream ==# 'stderr'
+            " Read from stderr instead of stdout.
+            let job = jobstart(command, {
+            \   'on_stderr': 'ale#engine#GatherOutputNeoVim',
+            \   'on_exit': 'ale#engine#HandleExitNeoVim',
+            \})
+        elseif a:linter.output_stream ==# 'both'
+            let job = jobstart(command, {
+            \   'on_stdout': 'ale#engine#GatherOutputNeoVim',
+            \   'on_stderr': 'ale#engine#GatherOutputNeoVim',
+            \   'on_exit': 'ale#engine#HandleExitNeoVim',
+            \})
+        else
+            let job = jobstart(command, {
+            \   'on_stdout': 'ale#engine#GatherOutputNeoVim',
+            \   'on_exit': 'ale#engine#HandleExitNeoVim',
+            \})
+        endif
+    else
+        let job_options = {
+        \   'in_mode': 'nl',
+        \   'out_mode': 'nl',
+        \   'err_mode': 'nl',
+        \   'close_cb': function('ale#engine#HandleExitVim'),
+        \}
+
+        if a:linter.output_stream ==# 'stderr'
+            " Read from stderr instead of stdout.
+            let job_options.err_cb = function('ale#engine#GatherOutputVim')
+        elseif a:linter.output_stream ==# 'both'
+            " Read from both streams.
+            let job_options.out_cb = function('ale#engine#GatherOutputVim')
+            let job_options.err_cb = function('ale#engine#GatherOutputVim')
+        else
+            let job_options.out_cb = function('ale#engine#GatherOutputVim')
+        endif
+
+        if has('win32')
+            " job_start commands on Windows have to be run with cmd /c,
+            " othwerwise %PATHTEXT% will not be used to programs ending int
+            " .cmd, .bat, .exe, etc.
+            let l:command = 'cmd /c ' . l:command
+        else
+            " On Unix machines, we can send the Vim buffer directly.
+            " This is faster than reading the lines ourselves.
+            let job_options.in_io = 'buffer'
+            let job_options.in_buf = a:buffer
+        endif
+
+        " Vim 8 will read the stdin from the file's buffer.
+        let job = job_start(l:command, l:job_options)
+    endif
+
+    " Only proceed if the job is being run.
+    if has('nvim') || (job !=# 'no process' && job_status(job) ==# 'run')
+        let a:linter.job = job
+
+        " Store the ID for the job in the map to read back again.
+        let s:job_info_map[ale#engine#GetJobID(job)] = {
+        \   'linter': a:linter,
+        \   'buffer': a:buffer,
+        \   'output': [],
+        \}
+
+        if has('nvim')
+            " In NeoVim, we have to send the buffer lines ourselves.
+            let input = join(getbufline(a:buffer, 1, '$'), "\n") . "\n"
+
+            call jobsend(job, input)
+            call jobclose(job, 'stdin')
+        elseif has('win32')
```

Comment:
You'll have to rebase again and incorporate the changes on master, although they may be temporary. There is an open issue with sending buffer data to jobs in MacVim GUI, but I believe the issue might be down to old versions of MacVim, so I might apply this only for the Windows version of Vim here in the end anyway.